source/routing/loading-and-error-substates.md

```diff
@@ -187,12 +187,73 @@ will look for:
 If one of the above is found, the router will immediately transition into
 that substate (without updating the URL). The "reason" for the error
 (i.e. the exception thrown or the promise reject value) will be passed
-to that error state as its `model`.
+to that error state as its `model`. For example, here we are rejecting a
+promise with the HTTP response code as the error message:
+
+```js
+  reject({"message": response.status})
+```
+
+`message` is an arbitrary key and you can use whatever is appropriate for your application.
+It is important that the reject contains an `object`. If you only reject with a textual value
+or number, for example a response code, then the reason will not bubble up to the route and 
+the model will be `null`. The following would result in a `null` model:
+
+```js
+  reject(404) // don't do this
+```
+
+This error "reason" will be passed to the error route available to the app and
+can then be set as the error controllers model via the routes `setupController` method. You can then
+display specific error messages in the error template based on this error "reason". You can
+use the appropriate error controller eg. `ErrorController` to display a message based on the 
+error message properties. For example,
+
+```app/router.js
+App.ErrorRoute = Ember.Route.extend({
+  setupController: function(controller, model, params) {
+    controller.set('model', model);
+  }
+});
+```
+
+```app/controllers/errorController.js
+App.ErrorController = Ember.ObjectController.extend({
```

Comment:
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

Hello,

I'm guessing you want the Ember CLI type stuff like

export default Ember.Route.extend({

??

Brave new world, haven't even begun to get my head round it all yet....

Cheers,

Ian

On 30/03/15 14:59, Trek Glowacki wrote:
>> +This error "reason" will be passed to the error route available
>> to the app and +can then be set as the error controllers model
>> via the routes `setupController` method. You can then +display
>> specific error messages in the error template based on this error
>> "reason". You can +use the appropriate error controller eg.
>> `ErrorController` to display a message based on the +error
>> message properties. For example, + +```app/router.js 
>> +App.ErrorRoute = Ember.Route.extend({ +  setupController:
>> function(controller, model, params) { +
>> controller.set('model', model); +  } +}); +``` + 
>> +```app/controllers/errorController.js +App.ErrorController =
>> Ember.ObjectController.extend({
> 
> No globals please.
> 
> --- Reply to this email directly or view it on GitHub: 
> https://github.com/emberjs/guides/pull/141/files#r27391826
> 
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1

iQEcBAEBAgAGBQJVGViyAAoJEPK45GBX+Cy5zLQH/2BCZLe7HZ2FySbgNYaVQ1Q8
Ke7i5r4lhLKwgAtDtqYlmogI2smUx5ejExiFWAPeyILkH94rsoa49Hjm/RMeJuEo
/XeEEoJ50xwSCNq5oWtjG18S6DzZw0nlDSZYMZsItuQ647vCyFP5jfKwMEi8qAQu
TOrKhSBWgjHgjCJr88oeJMqkpImuoijmNofthcHf5bey0kJiamJwce+Hpqd5xbns
AUVoNjqtk7VkfRyX8AP2WXm+WVOdin47ewNfqPA+IKI4b+tlCfqF7vrRhGaUCera
CCe1eY3O4tw2lb6tk3XPZp9HMpXHCcouxTUqFhct+nhEmEFFbaMv30e8F6+BQJo=
=VpYA
-----END PGP SIGNATURE-----